crates/forge/bin/cmd/eip712.rs

```diff
@@ -0,0 +1,241 @@
+use clap::{Parser, ValueHint};
+use eyre::{Ok, OptionExt, Result};
+use foundry_cli::{opts::CoreBuildArgs, utils::LoadConfig};
+use foundry_common::compile::ProjectCompiler;
+use foundry_compilers::{
+    artifacts::{
+        output_selection::OutputSelection,
+        visitor::{Visitor, Walk},
+        ContractDefinition, EnumDefinition, SourceUnit, StructDefinition, TypeDescriptions,
+        TypeName,
+    },
+    CompilerSettings,
+};
+use std::{collections::BTreeMap, path::PathBuf};
+
+foundry_config::impl_figment_convert!(Eip712Args, opts);
+
+/// CLI arguments for `forge eip712`.
+#[derive(Clone, Debug, Parser)]
+pub struct Eip712Args {
+    /// The path to the file from which to read struct definitions.
+    #[arg(value_hint = ValueHint::FilePath, value_name = "PATH")]
+    pub target_path: PathBuf,
+
+    #[command(flatten)]
+    opts: CoreBuildArgs,
+}
+
+impl Eip712Args {
+    pub fn run(self) -> Result<()> {
+        let config = self.try_load_config_emit_warnings()?;
+        let mut project = config.create_project(false, true)?;
+        let target_path = dunce::canonicalize(self.target_path)?;
+        project.settings.update_output_selection(|selection| {
```

Comment:
If we're only reading one file can we just set the project to compile that?

Reply:
we're doing `.files([target_path.clone()])` below, or you mean something else?

Reply:
missed that